platform/features/events/src/EventListController.js

```diff
@@ -0,0 +1,133 @@
+ * under the License.
+ *
+ * Open MCT Web includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+/**
+ * Module defining EventListController. 
+ * Created by chacskaylo on 06/18/2015. 
+ * Modified by shale on 06/23/2015.
+ */
+define(
+	["./DomainColumn", "./RangeColumn", "./EventListPopulator"],
+	function (DomainColumn, RangeColumn, EventListPopulator) {
+		"use strict";
+        
+		var ROW_COUNT = 100;
+
+		/**
+		 * The EventListController is responsible for populating
+		 * the contents of the event list view.
+		 * @constructor
+		 */
+		function EventListController($scope, formatter) {
+			var populator;
+            
+			// Get a set of populated, ready-to-display rows for the
+			// latest data values.
+			function getRows(telemetry) {
+				var datas = telemetry.getResponse(),
+					objects = telemetry.getTelemetryObjects();
+                
+				return populator.getRows(datas, objects, ROW_COUNT);
+			}
+
+			// Update the contents
+			function updateRows() {
+				var telemetry = $scope.telemetry;
+				$scope.rows = telemetry ? getRows(telemetry) : [];
+			}
+
+			// Set up columns based on telemetry metadata. This will
+			// include one column for each domain and range type, as
+			// well as a column for the domain object name.
+			function setupColumns(telemetry) {
+				var domainKeys = {},
+					rangeKeys = {},
+					columns = [],
+					metadata;
+
+				// Add a domain to the set of columns, if a domain
+				// with the same key has not yet been inclued.
+				function addDomain(domain) {
+					var key = domain.key;
+					if (key && !domainKeys[key]) {
+						domainKeys[key] = true;
+						columns.push(new DomainColumn(domain, formatter));
+					}
+				}
+
+                // Add a range col to the set of columns, if a range
+				// with the same key has not yet been included.
+				function addRange(range) {
+					var key = range.key;
+					if (key && !rangeKeys[key]) {
+						rangeKeys[key] = true;
+						columns.push(new RangeColumn(range, formatter));
+					}
+				}
+                
+				// We cannot proceed if the telemetry controller
+				// is not available; clear all rows/columns.
+				if (!telemetry) {
+					columns = [];
+					$scope.rows = [];
+					$scope.headers = [];
+					return;
+				}
+
+				// Add domain, range, event msg columns
+				metadata = telemetry.getMetadata();
+				(metadata || []).forEach(function (metadata) {
+					(metadata.domains || []).forEach(addDomain);
+				});
+				(metadata || []).forEach(function (metadata) {
+					(metadata.ranges || []).forEach(addRange);
+				});
+                
+                // Add default domain and range columns if none
```

Comment:
Looks like whitespace here (and in comment above) is off a little bit. May need to replace tabs with spaces